Mathlib/Algebra/Ring/BooleanRing.lean

```diff
@@ -0,0 +1,623 @@
+/-
+Copyright (c) 2021 Bryan Gin-ge Chen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bryan Gin-ge Chen, Yaël Dillies
+
+! This file was ported from Lean 3 source module algebra.ring.boolean_ring
+! leanprover-community/mathlib commit 4c19a16e4b705bf135cf9a80ac18fcc99c438514
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.PUnitInstances
+import Mathlib.Tactic.Abel
+import Mathlib.Tactic.Ring
+import Mathlib.Order.Hom.Lattice
+import Mathlib.Tactic.ScopedNS
+
+/-!
+# Boolean rings
+
+A Boolean ring is a ring where multiplication is idempotent. They are equivalent to Boolean
+algebras.
+
+## Main declarations
+
+* `BooleanRing`: a typeclass for rings where multiplication is idempotent.
+* `BooleanRing.toBooleanAlgebra`: Turn a Boolean ring into a Boolean algebra.
+* `BooleanAlgebra.ToBooleanRing`: Turn a Boolean algebra into a Boolean ring.
+* `AsBoolalg`: Type-synonym for the Boolean algebra associated to a Boolean ring.
+* `AsBoolring`: Type-synonym for the Boolean ring associated to a Boolean algebra.
```

Comment:
Presumably should at least be `AsBoolAlg` and `AsBoolRing`. Not sure why these weren't just `AsBooleanAlgebra` and `AsBooleanRing` in the first place... :-(

Reply:
because those names are unwieldingly long. Is that a problem for the port?

Reply:
changed them to `AsBoolAlg` and `AsBoolRing` as @semorrison suggested.